src/components/Modal.tsx

```diff
@@ -48,23 +51,25 @@ export const Modal = (props: {
   );
 };
 
-const useBodyRoot = () => {
+const useBodyRoot = (theme: AppState["theme"]) => {
   const [div, setDiv] = useState<HTMLDivElement | null>(null);
 
   const isMobile = useIsMobile();
   const isMobileRef = useRef(isMobile);
   isMobileRef.current = isMobile;
 
+  const excalidrawContainer: HTMLDivElement | null = useExcalidrawContainer();
+
   useLayoutEffect(() => {
     if (div) {
       div.classList.toggle("excalidraw--mobile", isMobile);
     }
   }, [div, isMobile]);
 
   useLayoutEffect(() => {
-    const isDarkTheme = !!document
-      .querySelector(".excalidraw")
-      ?.classList.contains("theme--dark");
+    const isDarkTheme =
+      !!excalidrawContainer?.classList.contains("theme--dark") ||
+      theme === "dark";
```

Comment:
if we don't want to use `theme` only (since it's optional), it should nevertheless take precedence since it's explicit, so let's swap the order of checks

Reply:
unclear what you mean, theme is given less precedence since its optional over excal container so why we want to check theme first ?

Reply:
if it's supplied and for some reason isn't matching the container, we use prefer the explicit value. Is there actually a reason why we're supplying it at all? Aren't Modals always part of the component tree?

Reply:
But I suppose since this isn't public API it doesn't matter, and actually is preferable UX-wise. It'll just mean the app might have slight bugs without us knowing.

Reply:
Yes, this isn't public API, if in the future we expose Dialog then it can be helpful when for the host but I don't think we will be doing that as the host will always have its own dialogs.